portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -54,11 +53,36 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       Comment comment = new Comment(id, name, text, timestamp);
       comments.add(comment);
     }
+    return comments;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    JsonObject jsonResponse = new JsonObject();
+    String maxCommentsString = request.getParameter("max-comments");
+
+    try {
+      int maxComments = Integer.parseInt(maxCommentsString);
+      List<Comment> comments = queryComments(maxComments);
+
+      Gson gson = new Gson();
+
+      JsonParser parser = new JsonParser();
+      JsonElement commentsJson = parser.parse(gson.toJson(comments));
 
-    Gson gson = new Gson();
+      jsonResponse.add("comments", commentsJson);
+      jsonResponse.addProperty("success", true);
+      response.setStatus(HttpServletResponse.SC_OK);
+      } catch (NumberFormatException nfe) {
+        jsonResponse.addProperty("success", false);
```

Comment:
I don't think this should be part of the final response. The success and failure is already indicated by the HTTP layer, there is no need for the message to also have this information.